src/schemaType/ArrayType.ts

```diff
@@ -37,7 +37,11 @@ class ArrayType extends BaseScalarArrayType {
 	public async validate(value: unknown, document: Document): Promise<string[]> {
 		return (
 			await Promise.all(
-				ensureArray(value).map((arrayItem) => this.valueSchemaType.validate(arrayItem, document)),
+				ensureArray(value).map(async (arrayItem, index) => {
+					const result = await this.valueSchemaType.validate(arrayItem, document);
+
+					return result.map((message) => `index ${index}: ${message}`);
```

Comment:
Rather than making a special message here, couldn't this return the array index as the map key representing the keyPath?

Reply:
Good idea.